### ISSUE TITLE
where upload function in demo?

### ISSUE THREAD

**Issue by abodacs** [danialfarid/angular-file-upload#523]
ng-file-change="upload(files)" in demo  i cannot find it in your demo and i can not use watch.
Thanks

**Comment by danialfarid** [comment]
it has to be `upload($files)`

**Comment by danialfarid** [comment]
the js code is here:
https://github.com/danialfarid/angular-file-upload/blob/master/demo/war/js/upload.js

**Comment by danialfarid** [comment]
Actually you are right the demo does not have that function.
So if you don't want to watch you can do this
```
<div ng-file-select ng-file-chage="upload($files)">upload</div>

and in your controller

$scope.upload = function($files) {
   $upload.upload({...});
}
```

**Comment by abodacs** [comment]
doesn't work with meand i use https://github.com/formly-js/angular-formly to generate upload div. 
Any hint

**Comment by danialfarid** [comment]
Can you create a jsfiddle to show the problem?

**Comment by firecentaur** [comment]
I always get $upload is an unkown provider, upon searching, stackechange indicates the $upload has been renamed to FileUploader, however FileUploader.upload complains that upload is an invalid function http://stackoverflow.com/questions/25800833/angularjs-angular-file-upload-unknown-provider-uploadprovider-error

**Comment by danialfarid** [comment]
@firecentaur this is a separate issue.
Also the stackoverflow issue has an answer selected so I guess your issue is resolved.

**Comment by danialfarid** [comment]
fixed in version 3.0.0
